pkg/storage/utils/decomposedfs/spaces.go

```diff
@@ -157,6 +151,17 @@ func (fs *Decomposedfs) CreateStorageSpace(ctx context.Context, req *provider.Cr
 		return nil, err
 	}
 
+	space, err := fs.storageSpaceFromNode(ctx, n, "*", n.InternalPath(), false)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &provider.CreateStorageSpaceResponse{
+		Status: &v1beta11.Status{
+			Code: v1beta11.Code_CODE_OK,
+		},
+		StorageSpace: space,
+	}
```

Comment:
Oh, the response is created here.... That reminds me on the discussion that we had about "from where do we report errors?" - Maybe instead of just returning `nil, err` in the code flow before, we should return a proper response object with an error code set? (I haven't really checked the code if that would make sense here, you probably know).

Reply:
This is the storage driver, the errors are catched by the storage provider which converts them into proper responses with error codes. (application level errors)

Reply:
good point. Currently the CS3Apis are constructed in that way. The storageprovider is a Wrapper around the storageDrivers, which implement an internal interface.